### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Kate-Khon**
[DEMO LINK](https://kate-khon.github.io/react_todo-app/)

### CODE REVIEW

**Review by viktorlyt — CHANGES_REQUESTED** [review]
Well done!
Some points from me:

9.  ...It should be visible if there is at least 1 completed item in the list.
10.Hide everything except the input to add new todo if there are no todos. But not if todos are just filtered out.
11.Todo title can't be empty! If a user presses Enter when the title is empty, this todo should be removed.

**Comment by Kate-Khon** [comment]
> Well done! Some points from me:
> 
> 9. ...It should be visible if there is at least 1 completed item in the list.
>    10.Hide everything except the input to add new todo if there are no todos. But not if todos are just filtered out.
>    11.Todo title can't be empty! If a user presses Enter when the title is empty, this todo should be removed.

**Comment by Kate-Khon** [comment]
> Well done! Some points from me:
> 
> 9. ...It should be visible if there is at least 1 completed item in the list.
>    10.Hide everything except the input to add new todo if there are no todos. But not if todos are just filtered out.
>    11.Todo title can't be empty! If a user presses Enter when the title is empty, this todo should be removed.

**Review by viktorlyt — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,93 +1,219 @@
...
+  const [stateOfTodo, setStateOfTodo]
+    = useState<Todo>({ id: 0, state: TodoState.COMPLETED });
+  const [titleOfTodo, setTitleOfTodo]
+    = useState<Todo>({ id: 0, title: '', state: TodoState.ACTIVE });
+
+  const setTodoState = (todoState: Todo): void => {
+    const { id, state } = todoState;
+    const todoIndexToChange = todos.findIndex(todo => todo.id === id);
+
+    const callback = (prev: Todo[], todo: Todo, index: number): Todo[] => {
```
Comment (mykola-koval) [inline src/App.tsx:36]: maybe let's use a better name for this function

File: src/App.tsx [hunk 2]
```diff
@@ -1,93 +1,219 @@
...
+    const changedTodos = todos.reduce(callback, []);
+
+    setTodos(changedTodos);
+  };
+
+  const setTodoTitle = (todoTitle: Todo): void => {
+    const { id, title, state } = todoTitle;
+    const todoIndexToChange = todos.findIndex(todo => todo.id === id);
+
+    const callback = (prev: Todo[], todo: Todo, index: number): Todo[] => {
```
Comment (mykola-koval) [inline src/App.tsx:55]: same here

File: src/App.tsx [hunk 3]
```diff
@@ -1,93 +1,219 @@
...
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view2" />
-              <label htmlFor="toggle-view2">1234567890</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
+        {todos.length > 0 && (
```
Comment (mykola-koval) [inline src/App.tsx:110]:
```suggestion
        {todos.length && (
```

File: src/components/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      todoId = todos[todos.length - 1].id + 1;
+    }
+
+    const newTodo = {
+      id: todoId,
+      title: query.trim(),
+      state: TodoState.ACTIVE,
+    };
+
+    if (newTodo.title !== '') {
```
Comment (mykola-koval) [inline src/components/Header.tsx:28]:
```suggestion
    if (newTodo.title) {
```

**Review by mykola-koval — APPROVED** [review]

**Pull request closed without merging**